src/test/java/testapp/adapter/http/UserServiceTest.java

```diff
@@ -0,0 +1,194 @@
+
+        final UserService userService = new UserService(userRepository, accountRepository);
+
+        context.checking(new Expectations(){{
+            oneOf(userRepository).delete(1);
+            will(throwException(new UserException("")));
+        }});
+
+        Reply<?> response = userService.deleteUserById(1);
+        Reply<?> userNotFound = Reply.saying().badRequest();
+
+        assertThat(response, returns(userNotFound));
+    }
+
+    @Test
+    public void findUserById() throws Exception {
+        final UserRepository userRepository = context.mock(UserRepository.class);
+        final AccountRepository accountRepository = context.mock(AccountRepository.class);
+
+        final User user = new User(1, "ianis");
+        final UserService userService = new UserService(userRepository, accountRepository);
+
+        context.checking(new Expectations(){{
+            oneOf(userRepository).findById(1);
+            will(returnValue(user));
+        }});
+
+        Reply<?> response = userService.findUserById(1);
+        Reply<?> userFound = Reply.with(user).as(Json.class);
+
+        assertThat(response, returns(userFound));
+        assertThat(userFound, containsJson(
+                            aNewJson()
+                                .add("id", user.getId())
+                                .add("name", user.getName())
+                                ));
+    }
+
+    @Test
+    public void findUserByIdThatNotExist() throws Exception {
+        final UserRepository userRepository = context.mock(UserRepository.class);
+        final AccountRepository accountRepository = context.mock(AccountRepository.class);
+
+        final UserService userService = new UserService(userRepository, accountRepository);
+
+        context.checking(new Expectations(){{
+            oneOf(userRepository).findById(1);
+            will(throwException(new UserException("")));
+        }});
+
+        Reply<?> response = userService.findUserById(1);
+        Reply<?> userNotFound = Reply.saying().badRequest();
+
+        assertThat(response, returns(userNotFound));
+    }
+
+    @Test
+    public void findAllUsers() throws Exception {
+        final UserRepository userRepository = context.mock(UserRepository.class);
+        final AccountRepository accountRepository = context.mock(AccountRepository.class);
+
+        final User ianis = new User(1, "ianis");
+        final UserService userService = new UserService(userRepository, accountRepository);
+
+        context.checking(new Expectations(){{
+            oneOf(userRepository).findUsers();
+            will(returnValue(Lists.newArrayList(ianis)));
+        }});
+
+        List<User> userList = Lists.newArrayList(ianis);
+
+        Reply<?> response = userService.getUsers();
+        Reply<?> userCreated = Reply.with(userList).as(Json.class);
+
+        assertThat(response, returns(userCreated));
+        assertThat(userCreated, containsJson(
```

Comment:
yes, I sow you are testing the json contained, but you are not testing the response from your service here